### PR TITLE
Feature/461 POC validation for Angular form control integration

### PR DESCRIPTION
**Pull request by anna-lach**
#461 POC - Angular validation for SLDS form control web components (we will need some refactor in the future)

### CODE REVIEW

File: packages/angular/projects/angular-core/src/forms/checkbox/checkbox.directive.ts [hunk 1]
```diff
@@ -14,56 +21,47 @@ import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
...
-export class CheckboxDirective implements ControlValueAccessor {
-  // eslint-disable-next-line @typescript-eslint/no-empty-function, @typescript-eslint/no-explicit-any
-  onChange: (value: any) => void = () => {};
-  // eslint-disable-next-line @typescript-eslint/no-empty-function, @typescript-eslint/no-explicit-any
-  onTouched: () => any = () => {};
+export class CheckboxDirective extends FormControlElementDirective {
 
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  private _value: any;
+  #initialValue: string | undefined;
```
Comment (jpzwarte) [inline packages/angular/projects/angular-core/src/forms/checkbox/checkbox.directive.ts:34]: Should this not be `#initialValue?: string;`

File: packages/angular/projects/angular-core/src/forms/checkbox/checkbox.directive.ts [hunk 2]
```diff
@@ -14,56 +21,47 @@ import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
...
 
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  get value(): any {
-    return this._value;
+  #value: string | undefined;
```
Comment (jpzwarte) [inline packages/angular/projects/angular-core/src/forms/checkbox/checkbox.directive.ts:36]: Same here

**Review by jpzwarte — APPROVED** [review]

File: packages/angular/projects/angular-core/src/forms/checkbox/checkbox.directive.ts (hunk 1 above)
Comment (anna-lach) [reply packages/angular/projects/angular-core/src/forms/checkbox/checkbox.directive.ts:34]: The value for the checkbox web component can be string or undefined, that's why I have it here string or undefined, to have the validation working properly.

File: packages/angular/projects/angular-core/src/forms/forms.module.ts [hunk 3]
```diff
@@ -1,4 +1,6 @@
 import { CUSTOM_ELEMENTS_SCHEMA, NgModule } from '@angular/core';
+import '@sl-design-system/shared';
```
[the remark this answers is not in the document]
Comment (anna-lach) [reply packages/angular/projects/angular-core/src/forms/forms.module.ts:2]: It's not necessary :)

File: packages/angular/.storybook/preview.ts [hunk 4]
```diff
@@ -1,17 +1,53 @@
+import { Preview } from '@storybook/angular';
 import '@webcomponents/scoped-custom-element-registry/scoped-custom-element-registry.min.js';
 import 'element-internals-polyfill';
+import {configureLocalization} from '@lit/localize';
 import { INITIAL_VIEWPORTS } from '@storybook/addon-viewport';
-import { Preview } from '@storybook/angular';
+import { updateTheme, themes } from './themes';
```
Comment (Diaan) [inline packages/angular/.storybook/preview.ts:6]: in chromatic i imported the themes from the "main" themes.ts, so we don't have 2 identical copies of this file (which will be a pain to maintain).
Have you tried if that is possible here as well?

**Review by Diaan — APPROVED** [review]

File: packages/angular/projects/angular-core/src/forms/input/input.directive.ts [hunk 5]
```diff
@@ -9,56 +21,44 @@ import {ControlValueAccessor, NG_VALUE_ACCESSOR} from "@angular/forms";
...
 
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  private _value: any;
+  #value?: string;
 
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  get value(): any {
-    return this._value;
+  get value(): string {
+    return this.#value ? this.#value : '';
```
Comment (jpzwarte) [inline packages/angular/projects/angular-core/src/forms/input/input.directive.ts:38]: `return this.#value ?? '';`

**Pull request merged**